package.json

```diff
@@ -6,16 +6,19 @@
   "repository": "github.com:formulanano/nanocar-builder-web.git",
   "license": "BSD-3-Clause",
   "dependencies": {
-    "@material-ui/core": "^4.4.3",
-    "@material-ui/icons": "^4.4.3",
-    "node-sass": "^4.12.0",
+    "@material-ui/core": "^4.6.1",
+    "@material-ui/icons": "^4.5.1",
+    "lodash": "^4.17.15",
+    "node-sass": "^4.13.0",
     "prop-types": "^15.7.2",
-    "react": "^16.9.0",
-    "react-dom": "^16.9.0",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
     "react-fps-stats": "^0.1.2",
-    "react-scripts": "^3.1.2",
-    "three": "^0.108.0",
-    "three-orbitcontrols": "^2.108.1"
+    "react-redux": "^7.1.3",
+    "react-scripts": "^3.2.0",
+    "redux": "^4.0.4",
```

Comment:
Is there a special reason for the redux decision?

Reply:
As UI grows in complexity, reaching to the state of components far apart in the component tree will be difficult. This makes the state difficult to maintain and less predictable. It also means passing unnecessary data to the components until you reach the target component.

With Redux, you’ve got a store where you can keep all your application state. If a state change occurs in Component A, it is then relayed to the central store and other components B and C can get aware of the change.

Let me give couple of examples that helped on taking the decision;

For example; we will need to have access to the state of current nanocar data on Viewport. We will need some components to be aware of some setting changes. Also we will need certain actions such as showing an information box or opening a menu to happen when you click to an atom or a molecule.

You can check out the UI design to understand the decision better.

Reply:
Sorry, I know what redux doing. Let me correct my question. Can't we use `react hooks` and `context` or `Mobx` instead of redux?

Reply:
Sure it is possible to use **React Context API**, **MobX** or maybe even **Flux** (please no 😄) but let me explain by the key points that helped me on my decision.

MobX is a good simple state management library with more than one store for data storage and mainly used for small and simple applications and not scalable as Redux.
As I said before, we expect application to grow in complexity so we will need that scalability that Redux provides out of the box.

Another key point is that Redux store is pure and quickly let you revert state updates. However, if it’s not done right, it's harder to debug MobX state updates.

Redux also uses Context API internally but the advantage of Redux in respect to Context API and MobX that it has a single source of truth for the entire application state. With that some functionality which is difficult to implement - **Undo/Redo,** for example - can suddenly become trivial to implement, if all of your state is stored in a single tree.

Therefore; I like MobX and Context API especially we might use Context API if it's necessary but looks like Redux is a way better fit to the project right now.